### PR TITLE
Get positions from Rhino to Quark

### PR DESCRIPTION
**Pull request by rachitnigam**
Propagating lines and column numbers from Rhino to Quark. 
- Made CNull and CUndefined objects. Changed serialization to have them as JsonProduct0.
- Made SNull, SEmptyExpression, SEmptyStatement objects.
- Changed Rhino to  quark so that the position are being set properly. There might be some issues with it that were fixed while debugging.
- Added tests from BrowserTestSuite to check that the positions are set correctly.

### CODE REVIEW

**Comment by rachitnigam** [comment]
The test that is failing was failing on the master branch when I checked it out. I don't believe it is something that is something that is caused from my code.

**Comment by arjunguha** [comment]
Bring it up to date with master and the tests should pass. (We have conflicts anyway.)

**Comment by rachitnigam** [comment]
Rebased and Fixed.

File: js/src/main/scala/JavaScriptCEKLike.scala [hunk 1]
```diff
@@ -420,11 +442,11 @@ trait JavaScriptCEKLike extends slogging.LazyLogging  {
       case Get(obj, key) => EConfig(obj, env, KGet(key, env) :: k)
       case Update(obj, key, e) =>  EConfig(obj, env, KUpdateR(key, e, env) :: k)
       case Declassify(e) => {
-        assert(thisLocation == Server)
+        assert(thisLocation == Server, "Declassifing only allowed on Server")
```
Comment (arjunguha) [inline js/src/main/scala/JavaScriptCEKLike.scala]: I suggest doing these kinds of refactorings in a separate pull request, but we'll take it.

File: shared/src/main/scala/Position.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	case class PositionException(msg: String) extends Exception(msg)
+
+	trait WithPos {
+		private var line: Option[Int] = None
+		private var column: Option[Int] = None
+
+		def setPos(nline: Option[Int], ncolumn: Option[Int]): Unit = {
+			if(nline.isDefined == false || ncolumn.isDefined == false)
+				println(Console.RED + s"Setting $this with $nline and $ncolumn" + Console.RESET)
```
Comment (arjunguha) [inline shared/src/main/scala/Position.scala]: @rachitnigam a println! Is this going to actually print? Would you remove it please.

**Comment by arjunguha** [comment]
LGTM

Would you remove the println and then do "squash and merge" (not merge).

**Pull request merged**